ts/session/types/PubKey.ts

```diff
@@ -84,18 +84,29 @@ export class PubKey {
     return typeof value === 'string' ? new PubKey(value) : value;
   }
 
-  public static shorten(value: string | PubKey): string {
+  public static shorten(
+    value: string | PubKey,
+    opts?: { keepCharacters?: number; withParenthesis?: boolean }
+  ): string {
     const valAny = value as PubKey;
     const pk = value instanceof PubKey ? valAny.key : value;
     if (!pk) {
       throw new Error('PubKey.shorten was given an invalid PubKey to shorten.');
     }
 
-    if (pk.length < 8) {
+    const keepCharacters = opts?.keepCharacters ?? 4;
+    const withParenthesis = opts?.withParenthesis ?? true;
+
+    if (pk.length < keepCharacters * 2) {
```

Comment:
Should this be `<=` because if keepChars is 3 and the name is 6 chars we don't want it to be shortened, should only shorten if the name is greater than max chars. Eg if the name is potato we dont want to turn that into pot...ato